packages/sdk/src/swap/v2/types.ts

```diff
@@ -3,149 +3,145 @@ import { SwapType } from '@/swap/client';
-  }
-  type SwapState =
-    | ({
-        state: 'RECEIVING';
-      } & Receiving)
-    | ({
-        state: 'SWAPPING';
-      } & Swapping)
-    | ({
-        state: 'SENDING';
-      } & Sending)
-    | ({
-        state: 'SENT';
-      } & Sending)
-    | ({
-        state: 'COMPLETED';
-      } & Sending)
-    | ({
-        state: 'FAILED';
-      } & Sending);
+  };
+}
+interface Boost {
+  maxBoostFeeBps: number;
+  effectiveBoostFeeBps: number | undefined;
+  boostedAt: number | undefined;
+  boostedBlockIndex: string | undefined;
+  skippedAt: number | undefined;
+  skippedBlockIndex: string | undefined;
+}
+interface DepositChannelFields {
+  createdAt: number;
+  brokerCommissionBps: number;
+  depositAddress: string;
+  srcChainExpiryBlock: string;
+  estimatedExpiryTime: number;
+  expectedDepositAmount: string;
+  isExpired: boolean;
+  openedThroughBackend: boolean;
+  affiliateBrokers: AffiliateBroker[];
+  fillOrKillParams: FillOrKillParams | undefined;
+  dcaParams: DcaParameters | undefined;
+  srcChainRequiredBlockConfirmations: number | null | undefined;
+}
+interface DepositFields {
+  amount: string | undefined;
+  txRef: string | undefined;
+  txConfirmations: number | undefined;
+  receivedAt: number | undefined;
+  receivedBlockIndex: string | undefined;
+  failure: Failure | undefined;
+}
+interface SwapChunk {
+  swapInputAmount: string;
+  swapOutputAmount: string | undefined;
+  scheduledAt: number;
+  scheduledBlockIndex: string;
+  executedAt: number | undefined;
+  executedBlockIndex: string | undefined;
+  retryCount: number;
+  latestSwapRescheduledAt: number | undefined;
+  latestSwapRescheduledBlockIndex: number | undefined;
+  fees: SwapFee[];
+}
+interface SwapFields {
+  totalInputAmountSwapped: string | undefined;
+  totalOutputAmountSwapped: string | undefined;
+  totalChunksExecuted: number;
+  isDcaSwap: boolean;
+  lastExecutedChunk: SwapChunk | undefined;
+  currentChunk: SwapChunk;
+  type: SwapType;
+  fees: SwapFee[];
+}
+interface EgressFields {
+  amount: string;
+  scheduledAt: number;
+  scheduledBlockIndex: string;
+  sentTxRef: string | undefined;
+  sentAt: number | undefined;
+  sentAtBlockIndex: string | undefined;
+  ignoredAmount: string | undefined;
+  failure: Failure | undefined;
+}
+interface SwapStatusResponseCommonFields extends ChainsAndAssets {
+  swapId: string;
+  destAddress: string;
+  depositChannel: DepositChannelFields;
+  ccm: Ccm | undefined;
+  boost: Boost | undefined;
+  estimatedDurationSeconds: number | null | undefined;
+}
 
-  export type DepositAddressStatusResponse = SwapState;
-  export type VaultSwapStatusResponse = SwapState & SendingVaultSwap;
+interface VaultSwapCommonFields
+  extends Exclude<SwapStatusResponseCommonFields, 'depositChannel' | 'boost'> {}
+interface ReceivingVaultSwap extends VaultSwapCommonFields {
+  deposit: Pick<DepositFields, 'amount' | 'txRef'>;
+}
+interface SwappingVaultSwap extends ReceivingVaultSwap {
```

Comment:
not sure if its worth it to have the separate `VaultSwap` types. could we just make the `depositChannel` optional?